app/Media/WEBP.php

```diff
@@ -0,0 +1,97 @@
+<?php
+namespace App\Media;
+
+use WebPConvert\WebPConvert;
+
+class WEBP
+{
+  private bool $convertImages = false;
+
+  private array $allowedMimes = [
+    'image/jpeg',
+    'image/png'
+  ];
+
+  /**
+   * @action init
+   */
+  public function action(): void
+  {
+    $this->convertImages = ! empty($option = firestarter()->settings()->get('site_webp_images')) ? $option : false;
+  }
+
+  private function convert(string $filePath): void
+  {
+    if (! empty($filePath) && $this->shouldGenerate($filePath)) {
+      try {
+        WebPConvert::convert($filePath, $this->getWebpPath($filePath), []);
+      } catch (\Throwable $th) {
+        error_log($th);
+      }
+    }
+  }
+
+  /**
+   * @filter wp_handle_upload 10
+   */
+  public function handleUpload(array $fileArray): array
+  {
+    $this->convert($fileArray['file']);
+    return $fileArray;
```

Comment:
Please fix formatting. There should be empty line before `return` statement everywhere.